Docs: Document portable stories usage (#28452)
storybookjs/storybook: docs/api/portable-stories/portable-stories-jest.mdx

  ## setProjectAnnotations

  This API should be called once, before the tests run, typically in a [setup file](https://jestjs.io/docs/configuration#setupfiles-array). This will make sure that whenever `composeStories` or `composeStory` are called, the project annotations are taken into account as well.
  
  These are the configurations needed in the setup file:
  - preview annotations: those defined in `.storybook/preview.ts`
  - addon annotations (optional): those exported by addons

>   - beforeAll: code that runs before all tests

Author: Would be great to point this to the `beforeAll` documentation, which we unfortunately do not have yet. I'm adding this comment here as a reminder for the future

Reviewer: Docs for `beforeAll` were added in https://github.com/storybookjs/storybook/pull/28444, so I added a link. 🎉

  ## Story pipeline

  To preview your stories in Storybook, Storybook runs a story pipeline, which includes applying project annotations, loading data, rendering the story, and playing interactions. This is a simplified version of the pipeline:

>   ![A flow diagram of the story pipeline. First, set project annotations. Collect annotations (decorators, args, etc) which are exported by addons and the preview file. Second, compose story. Create renderable elements based on the stories passed onto the API. Third, render story. Load, mount, and execute the play function as part of the portable stories API.](../../_assets/api/story-pipeline.png)

Author: The alt needs an update on the third step, it's not up to date to the image (same for the vitest one)
```suggestion
  ![A flow diagram of the story pipeline. First, set project annotations. Collect annotations (decorators, args, etc) which are exported by addons and the preview file. Second, compose story. Create renderable elements based on the stories passed onto the API. Third, play. Mount the component and execute all the story lifecycle hooks.](../../_assets/api/story-pipeline.png)
```
